### DISCUSSION TITLE
Lesson 11: DeployBasicNft.s.sol and Interactions.s.sol

### DISCUSSION THREAD

**Discussion by 0xPublicGoods** [Cyfrin/foundry-full-course-cu#58] [Q&A] [answered]
Hey Team

---edited---
to include that this is when i am running on anvil, my git repo run-latest.json shows when i attempted to do sepolia but this didnt verify the contract nor was i able to mint - error looked the same but i didnt have a chance to copy paste it
---fin edit---

So I've just loaded everything to git, my run-latest.json shows I have deployed a contract.

But on running make mint I get the following error log

```bash
== Logs ==
  Return Data:
  0x0000000000000000000000000000000000000000
Error: 
No contract deployed
```
Full Response from : make mint

```bash
➜  foundry-nft-f23 git:(main) make mint
[⠆] Compiling...
No files changed, compilation skipped
2023-06-12T20:49:04.661954Z ERROR apply:ext: evm::cheatcodes: non-empty stderr args=["bash", "/Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/lib/foundry-devops/src/get_recent_deployment.sh", "BasicNft", "31337", "/Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23//broadcast"] stderr="/Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 31: jq: command not found\n/Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 33: jq: command not found\n/Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 35: ((: i<: syntax error: operand expected (error token is \"<\")\n"
Traces:
  [639098] → new DevOpsTools@0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9
    └─ ← 3192 bytes of code

  [295357] → new MintBasicNft@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← 1254 bytes of code

  [59970] MintBasicNft::run() 
    ├─ [35980] DevOpsTools::e374cdf1(00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000007a69000000000000000000000000000000000000000000000000000000000000000842617369634e6674000000000000000000000000000000000000000000000000) [delegatecall]
    │   ├─ [0] VM::ffi([pwd]) 
    │   │   └─ ← 0x2f55736572732f627261646c65796d6f6f72652f436f64652f30785075626c6963476f6f64732f666f756e6472792d6632332f666f756e6472792d6e66742d663233
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← 0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000053331333337000000000000000000000000000000000000000000000000000000
    │   ├─ [0] VM::ffi([bash, /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/lib/foundry-devops/src/get_recent_deployment.sh, BasicNft, 31337, /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23//broadcast]) 
    │   │   └─ ← 0x0000000000000000000000000000000000000000
    │   ├─ [0] console::log(Return Data:) [staticcall]
    │   │   └─ ← ()
    │   ├─ [0] console::log(0x0000000000000000000000000000000000000000) [staticcall]
    │   │   └─ ← ()
    │   └─ ← 0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000144e6f20636f6e7472616374206465706c6f796564000000000000000000000000
    └─ ← "No contract deployed"



== Logs ==
  Return Data:
  0x0000000000000000000000000000000000000000
Error: 
No contract deployed

make: *** [mint] Error 1
```

**Comment by 0xPublicGoods** [comment]
My Deploy Output from the Console, prior to make mint, is as follows:

: make deploy

```bash
➜  foundry-nft-f23 git:(main) make deploy
[⠆] Compiling...
No files changed, compilation skipped
Script ran successfully.

== Return ==
0: contract BasicNft 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0

EIP-3855 is not supported in one or more of the RPCs used.
Unsupported Chain IDs: 31337.
Contracts deployed with a Solidity version equal or higher than 0.8.20 might not work properly.
For more information, please see https://eips.ethereum.org/EIPS/eip-3855

## Setting up (1) EVMs.

==========================

Chain 31337

Estimated gas price: 4.7696455 gwei

Estimated total gas used for script: 1531904

Estimated amount required: 0.007306639020032 ETH

==========================

###
Finding wallets for all the necessary addresses...
##
Sending transactions [0 - 0].
⠁ [00:00:00] [##############################################################################################################] 1/1 txes (0.0s)
Transactions saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/broadcast/DeployBasicNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/cache/DeployBasicNft.s.sol/31337/run-latest.json

##
Waiting for receipts.
⠉ [00:00:00] [##########################################################################################################] 1/1 receipts (0.0s)
##### anvil-hardhat
✅ Hash: 0xf82be7aa4508bc913b42d51dab9ec124e0dd596dff160a4f4f9ea351707f82d4
Contract Address: 0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0
Block: 3
Paid: 0.00445903103547027 ETH (1178730 gas * 3.782911299 gwei)


Transactions saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/broadcast/DeployBasicNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/cache/DeployBasicNft.s.sol/31337/run-latest.json



==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.00445903103547027 ETH (1178730 gas * avg 3.782911299 gwei)

Transactions saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/broadcast/DeployBasicNft.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/0xPG/Code/0xPublicGoods/foundry-f23/foundry-nft-f23/cache/DeployBasicNft.s.sol/31337/run-latest.json
```

**Reply by PatrickAlphaC** [reply, marked as answer]
Yep! You'll need to install [jq](https://stackoverflow.com/questions/37668134/how-to-install-jq-on-mac-on-the-command-line)

**Reply by stefannovak** [reply]
Cheers, fixed for me

**Reply by ShayanShamsi** [reply]
Thanks @PatrickAlphaC! Fixed it for me :)

For Ubuntu users, you can use apt-get to install jq:
```bash
sudo apt-get update
sudo apt-get -y install jq
```

**Reply by RichardPalladino** [reply]
woot woot.  thank you sir.

**Comment by musprodev** [comment]
I wanna talk to sir Patrick Collins on a big project. Will I be able to chat you privately 🙇‍♂️

**Reply by PatrickAlphaC** [reply]
If you're looking for an audit, hit us up at audits@cyfrin.io - otherwise please use the forums for questions! Thank you!

**Comment by ermalbujupi** [comment]
@PatrickAlphaC I'm getting the same error and I'm kinda stuck. I already installed jq.

2023-08-09T21:14:59.176232Z ERROR apply:ext: evm::cheatcodes: non-empty stderr args=["bash", "/Users/ermalbujupaj/Development/Decentralized Apps/foundry-nft/lib/foundry-devops/src/get_recent_deployment.sh", "BasicNFT", "11155111", "/Users/ermalbujupaj/Development/Decentralized Apps/foundry-nft//broadcast"] stderr="find: /Users/ermalbujupaj/Development/Decentralized: No such file or directory\nfind: Apps/foundry-nft//broadcast: No such file or directory\n"
Traces:
  [639098] → new DevOpsTools@0xaFEc79BddB6b0FcB600aC876fa7001692dc576FF
    └─ ← 3192 bytes of code

  [249714] → new MintBasicNft@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← 1026 bytes of code

  [59759] MintBasicNft::run() 
    ├─ [35769] DevOpsTools::e374cdf1(00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000aa36a7000000000000000000000000000000000000000000000000000000000000000842617369634e4654000000000000000000000000000000000000000000000000) [delegatecall]
    │   ├─ [0] VM::ffi([pwd]) 
    │   │   └─ ← 0x2f55736572732f65726d616c62756a7570616a2f446576656c6f706d656e742f446563656e7472616c697a656420417070732f666f756e6472792d6e6674
    │   ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   │   └─ ← 0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000083131313535313131000000000000000000000000000000000000000000000000
    │   ├─ [0] VM::ffi([bash, /Users/ermalbujupaj/Development/Decentralized Apps/foundry-nft/lib/foundry-devops/src/get_recent_deployment.sh, BasicNFT, 11155111, /Users/ermalbujupaj/Development/Decentralized Apps/foundry-nft//broadcast]) 
    │   │   └─ ← 0x0000000000000000000000000000000000000000
    │   ├─ [0] console::log(Return Data:) [staticcall]
    │   │   └─ ← ()
    │   ├─ [0] console::log(0x0000000000000000000000000000000000000000) [staticcall]
    │   │   └─ ← ()
    │   └─ ← 0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000144e6f20636f6e7472616374206465706c6f796564000000000000000000000000
    └─ ← "No contract deployed"



== Logs ==
  Return Data:
  0x0000000000000000000000000000000000000000
Error: 
No contract deployed

make: *** [mint] Error 1

**Reply by iftikharuddin** [reply]
Yap I am also getting this error, it's kinda very strange 

@PatrickAlphaC can you please take a look champ 🙏 

<img width="1380" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/10365184/6be0f2ac-ed19-44d3-8db7-59f947b20c16">

**Reply by RichardPalladino** [reply]
Are certain you successfully installed jq plus all the other dependencies?

**Reply by iftikharuddin** [reply]
yap i did installed **jq** successfully, what other dependencies you are referring to? Everything is synced.
